frontend/src/canvas/Canvas.js

```diff
@@ -35,58 +28,33 @@ const Canvas = props => {
       shouldReconnect: () => true,
     },
   )
-  
-  // TODO: Weird positioning behavior when clicking into devtools
-
-  // Handle wheel event for zooming
-  const handleWheel = (e) => {
-    let newScale = canvasScale
-    if (e.deltaY < 0) {
-      newScale = Math.min(maxScale, newScale + 0.2)
-    } else {
-      newScale = Math.max(minScale, newScale - 0.2)
-    }
-    // TODO: Smart positioning of canvas zoom ( zoom to center of mouse pointer )
-    //let newCanvasPositionX = canvasPositionX
-    //let newCanvasPositionY = canvasPositionY
-    //const canvasOriginX = canvasPositionX + width / 2
-    //const canvasOriginY = canvasPositionY + height / 2
-    //setCanvasPositionX(newCanvasPositionX)
-    //setCanvasPositionY(newCanvasPositionY)
 
-    setCanvasScale(newScale)
-  }
+  // TODO: Weird positioning behavior when clicking into devtools
+  useEffect(() => {
+    const canvas = select(canvasPositionRef.current)
+    const Dzoom = zoom().on("zoom", zoomHandler)
 
-  const handlePointerDown = (e) => {
-    setIsDragging(true)
-    setDragStartX(e.clientX)
-    setDragStartY(e.clientY)
-  }
+    // Set default zoom level and center the canvas
+    canvas
+      .call(Dzoom)
+      .call(Dzoom.transform, zoomIdentity.translate(0, 0).scale(4))
 
-  const handlePointerUp = () => {
-    setIsDragging(false)
-    setDragStartX(0)
-    setDragStartY(0)
-  }
+    return () => {
+      canvas.on(".zoom", null); // Clean up zoom event listeners
+    };
+  }, []);
 
-  const handlePointerMove = (e) => {
-    if (isDragging) {
-      // TODO: Prevent dragging outside of canvas container
-      setCanvasPositionX(canvasPositionX + e.clientX - dragStartX)
-      setCanvasPositionY(canvasPositionY + e.clientY - dragStartY)
-      setDragStartX(e.clientX)
-      setDragStartY(e.clientY)
-    }
+  const zoomHandler = (event) => {
+    const ele = canvasScaleRef.current
+    const {
+      k: newScale,
+      x: newCanvasPositionX,
+      y: newCanvasPositionY,
+    } = event.transform;
+    const transformValue = `translate(${newCanvasPositionX}px, ${newCanvasPositionY}px) scale(${newScale})`
```

Comment:
Can you set & check against min & max zoom scales? Hardcoded minScale = 1 & maxScale = 40 for now is good.

Reply:
Okay, i will make the requested changes.

Reply:
Added the requested changes kindly review